addons/base_address_extended/models/res_city.py

```diff
@@ -25,7 +25,8 @@ def name_get(self):
     @api.model
     def _name_search(self, name='', args=None, operator='ilike', limit=100, name_get_uid=None):
         args = list(args or [])
-        # optimize out the default criterion of ``ilike ''`` that matches everything
-        if not (name == '' and operator == 'ilike'):
-            args += ['|', (self._rec_name, operator, name), ('zipcode', operator, name)]
-        return self._search(args, limit=limit, access_rights_uid=name_get_uid)
+        domain = []
+        if name:
+            connector = '!' if operator in expression.NEGATIVE_TERM_OPERATORS else '|'
+            domain = [connector, ('zipcode', 'ilike', name), (self._rec_name, operator, name)]
+        return self._search(expression.AND([domain, args]), limit=limit, access_rights_uid=name_get_uid)
```

Comment:
plz comment this override to explain why you made it :)

Reply:
done